src/components/fields/submit-button/submit-button.tsx

```diff
@@ -24,11 +29,22 @@ export const SubmitButton = (props: IGenericFieldProps<ISubmitButtonSchema>) =>
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, []);
 
+	useDeepCompareEffect(() => {
```

Comment:
not very familiar with `useWatch`, why is a deep compare needed? (I did notice the object reference of `formValues` would change even when the same dropdown value was selected, but inputs were fine)

Reply:
hmm I am doing a deep compare to avoid re-rendering the button when the values are touched but not updated, like the dropdown interaction you cited